B 21569 int

stevengleason-caci opened the pull request:
# B-21569 INT
[B-21569](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1024477)
## Summary
Add service items for Unaccompanied Baggage. They are visible after the shipment is Approved and Submitted. They cane be Rejected and re-Approved.

The feature mentions that the Fuel Surcharge is only for when either the Origin or Destination is CONUS.
I updated the preview page also.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Run with feature flags on: FEATURE_FLAG_UNACCOMPANIED_BAGGAGE=true FEATURE_FLAG_ENABLE_ALASKA=true make server_run
2. Login as a Customer and create an OCONUS-to-CONUS UB shipment
3. And create an OCONUS-to-OCONUS UB shipment
4. Log in as a Counselor and complete and approve both shipments
5. Log in as a TOO and submit the OCONUS-to-CONUS shipment
6. See the 4 service items
7. Submit the OCONUS-to-OCONUS shipment
8. See only 3 service items (no Fuel Surcharge)

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### Database

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
### CONUS-OCONUS with POE FSC
![Screenshot 2025-01-07 at 1 13 55 PM](https://github.com/user-attachments/assets/e29df5d8-b43b-41b4-974b-f91fe96073e0)
![Screenshot 2025-01-07 at 1 14 22 PM](https://github.com/user-attachments/assets/f8191295-f234-4674-8363-f8c9a44105de)
### OCONUS-CONUS with POD FSC
![Screenshot 2025-01-07 at 1 15 05 PM](https://github.com/user-attachments/assets/1f56d97d-6205-4ff8-af76-02535d81e854)
![Screenshot 2025-01-07 at 1 15 35 PM](https://github.com/user-attachments/assets/df54aed7-3ca0-42e6-a900-2eaa9163331f)
### OCONUS-OCONUS (no FSC)![Screenshot 2025-01-07 at 1 16 04 PM](https://github.com/user-attachments/assets/909d2ec3-764a-46fd-9689-7725797e2b44)![Screenshot 2025-01-07 at 1 16 22 PM](https://github.com/user-attachments/assets/2c3e830a-b4bf-43a1-a0c2-31469de1ea1d)

ryan-mchugh commented:
I noticed that FSC still shows in the preview for OCONUS-OCONUS UB shipments

<img width="1187" alt="Screenshot 2024-12-31 at 3 37 20 PM" src="https://github.com/user-attachments/assets/6ee3dcfb-69d2-4e57-a528-9ce7af990512" />

stevengleason-caci commented:
I updated the preview page so AK to AK doesn't produce the Fuel Surcharge.
I also changed my code to use the DB instead of coding it in the switch-case.

danieljordan-caci commented on pkg/services/mto_shipment/shipment_approver_test.go:
```diff
@@ -854,4 +854,184 @@ func (suite *MTOShipmentServiceSuite) TestApproveShipment() {
+				Model: models.MTOShipment{
+					MarketCode:   "i",
+					Status:       models.MTOShipmentStatusSubmitted,
+					ShipmentType: "UNACCOMPANIED_BAGGAGE",
```
Can you use a variable here instead of hard coding `"UNACCOMPANIED_BAGGAGE"`?

Should be `models.MTOShipmentTypeUnaccompaniedBaggage`

danieljordan-caci commented on pkg/services/mto_shipment/shipment_approver_test.go:
```diff
@@ -854,4 +854,184 @@ func (suite *MTOShipmentServiceSuite) TestApproveShipment() {
+			},
+			{
+				Model: models.MTOShipment{
+					MarketCode:   "i",
```
Can you use the variable for this instead of hard coding?

`models.MarketCodeInternational`

danieljordan-caci commented on pkg/services/mto_shipment/shipment_approver_test.go:
```diff
@@ -854,4 +854,184 @@ func (suite *MTOShipmentServiceSuite) TestApproveShipment() {
+			},
+			{
+				Model: models.MTOShipment{
+					MarketCode:   "i",
+					Status:       models.MTOShipmentStatusSubmitted,
+					ShipmentType: "UNACCOMPANIED_BAGGAGE",
```
Same thing here with avoiding these hard coded strings when we have the variables available.

stevengleason-caci replied:
Updated

danieljordan-caci commented on pkg/services/mto_shipment/shipment_approver_test.go:
```diff
@@ -854,4 +854,184 @@ func (suite *MTOShipmentServiceSuite) TestApproveShipment() {
+			},
+			{
+				Model: models.MTOShipment{
+					MarketCode:   "i",
+					Status:       models.MTOShipmentStatusSubmitted,
+					ShipmentType: "UNACCOMPANIED_BAGGAGE",
```
Use variables plz

danieljordan-caci commented on src/components/Office/ShipmentServiceItemsTable/ShipmentServiceItemsTable.test.jsx:
```diff
@@ -390,4 +397,17 @@ describe('Shipment Service Items Table', () => {
   });
+
+  describe('renders the intl UB shipment type (OCONUS -> OCONUS) with service items', () => {
+    it.each([['International UB'], ['International UB pack'], ['International UB unpack']])(
```
Can you also add a test for CONUS -> OCONUS & OCONUS -> CONUS?

stevengleason-caci commented:
129 diffs in Happo: all the boxes have rounded corners. I don't think it's me. It's also showing up in these:
#14525 #14522 and others, and accepted in #14404

danieljordan-caci commented:
> 129 diffs in Happo: all the boxes have rounded corners. I don't think it's me. It's also showing up in these: #14525 #14522 and others, and accepted in #14404



> 129 diffs in Happo: all the boxes have rounded corners. I don't think it's me. It's also showing up in these: #14525 #14522 and others, and accepted in #14404

approved

brianmanley-caci reviewed (approved):
All 3 cases worked for me 👏
